packages/nextra/src/setup-page.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * ⚠️ Attention!
+ * This file should be never used directly, only in loader.ts
+ */
+
+import { FC } from 'react'
+import get from 'lodash.get'
+import { IS_PRODUCTION, NEXTRA_INTERNAL } from './constants'
+import {
+  DynamicMetaDescriptor,
+  NextraInternalGlobal,
+  PageOpts,
+  ThemeConfig
+} from './types'
+
+/**
+ * Calculate a 32 bit FNV-1a hash
+ * Found here: https://gist.github.com/vaiorabbit/5657561
+ * Ref.: http://isthe.com/chongo/tech/comp/fnv/
+ *
+ * @param {string} str the input value
+ * @param {number} [seed] optionally pass the hash of the previous chunk
+ * @returns {string}
+ */
+function hashFnv32a(str: string, seed = 0x811c9dc5): string {
+  let hval = seed
+
+  for (let i = 0; i < str.length; i++) {
+    hval ^= str.charCodeAt(i)
+    hval += (hval << 1) + (hval << 4) + (hval << 7) + (hval << 8) + (hval << 24)
+  }
+
+  // Convert to 8 digit hex string
+  return ('0000000' + (hval >>> 0).toString(16)).substring(-8)
+}
+
+export function setupNextraPage({
+  pageNextRoute,
+  pageOpts,
+  nextraLayout,
+  themeConfig,
+  Content,
+  hot,
+  dynamicMetaItems
+}: {
+  pageNextRoute: string
+  pageOpts: PageOpts
+  nextraLayout: FC
+  themeConfig: ThemeConfig
+  Content: FC
+  hot: __WebpackModuleApi.Hot
+  dynamicMetaItems: DynamicMetaDescriptor[]
+}) {
+  if (typeof window === 'undefined') {
+    globalThis.__nextra_resolvePageMap = async () => {
+      const clonedPageMap = JSON.parse(JSON.stringify(pageOpts.pageMap))
+      await Promise.all(
+        dynamicMetaItems.map(
+          async ({ metaFilePath, metaObjectKeyPath, metaParentKeyPath }) => {
+            const mod = await import(metaFilePath)
+            const metaData = await mod.default()
+            const meta = get(clonedPageMap, metaObjectKeyPath)
```

Comment:
@shuding if you'll refactor this code https://github.com/shuding/nextra/blob/5f5e06bb8257cc4a3fa533cd03927f855aaba222/packages/nextra/src/page-map.ts#L29 we can remove `lodash.get`